_rules/aria-required-accessible-name-gp8n89.md

```diff
@@ -0,0 +1,275 @@
+  <span aria-hidden="true">★★★★★</span>
+</div>
+```
+
+#### Passed Example 4
+
+The `div` element with role `checkbox` has an [accessible name][] provided thanks to the `aria-labelledby` attribute.
+
+```html
+<div role="checkbox" aria-checked="false" tabindex="0" aria-labelledby="pass-agree-tc"></div>
+<div id="pass-agree-tc">I agree with terms and conditions</div>
+```
+
+#### Passed Example 5
+
+The `div` element with role `dialog` has an [accessible name][] provided by the `aria-labelledby` attribute.
+
+```html
+<div role="dialog" aria-modal="true" tabindex="-1" aria-labelledby="pass-terms">
+  <h1 id="pass-terms">Terms</h1>
+  <p>These are our terms.</p>
+</div>
+```
+
+#### Passed Example 6
+
+The `div` element with role `heading` has an [accessible name][] provided by its content.
+
+```html
+  <div role="heading" aria-level="1">Terms</div>
+```
+
+#### Passed Example 7
+
+This `div` element with role `heading` is not [visible][], but is still [included in the accessibility tree][]. It has a non-empty [accessible name][] provided by its content.
+
+```html
+<div role="heading" aria-level="1" style="position: absolute; left: -9999px">ACT rules</div>
+```
+
+#### Passed Example 8
+
+The `div` element with role `link` has an [accessible name][] provided by its content.
+
+```html
+<div role="link" tabindex="0" onclick="location.href='https://act-rules.github.io/'">ACT Rules</div>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The `div` element with role `button` doesn't have an [accessible name][].
+
+```html
+<div role="button"></div>
+```
+
+#### Failed Example 2
+
+This `button` element with an [implicit semantic role] of `button` doesn't have an [accessible name][].
+
+```html
+<button></button>
+```
+
+#### Failed Example 3
+
+The `div` element with role `img` doesn't have an [accessible name][].
+
+```html
+<div role="img">
+  <span aria-hidden="true">★★★★★</span>
+</div>
+```
+
+#### Failed Example 4
+
+The `div` element with role `checkbox` doesn't have an [accessible name][].
+
+```html
+<div role="checkbox" aria-checked="false" tabindex="0"></div>
+<div>I agree with terms and conditions</div>
+```
+
+#### Failed Example 5
+
+The `dialog` role accepts an [accessible name][] only from the author and not from its content. Therefore, the `div` element with role `dialog` doesn't have an [accessible name][].
+
+```html
+<div role="dialog" aria-modal="true" tabindex="-1">
+  <h1>Terms</h1>
+  <p>These are our terms.</p>
+</div>
+```
+
+#### Failed Example 6
+
+The `div` element with role `heading` has an empty [accessible name][] due to the empty `aria-label` [attribute value][].
+
+```html
+  <div role="heading" aria-level="1" aria-label="">Terms</div>
+```
+
+#### Failed Example 7
+
+The `div` element with role `heading` has an empty [accessible name][] given by its `aria-labelledby` attribute.
+
+```html
+<span id="fail-heading"></span>
+<div role="heading" aria-level="1" aria-labelledby="fail-heading">ACT Rules</div>
+```
```

Comment:
I don't really like this example either. This rule is about checking that elements have accessible names. ACT rules aren't bug trackers to ensure tools have a flawless accessible name calculation.

Reply:
I'm not sure I understand the point. 

I agree with the previous point (Failed Example 6), as the AccName specifications indicate that empty aria-labels should be ignored. 

However, I consider this a clear failure, since the element with the role of heading lacks an accessible name due to improper handling by the author.